update master

ciband opened the pull request:
## Proposed changes
<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
-->

## Types of changes
<!--
What types of changes does your code introduce?
_Put an `x` in the boxes that apply and remove the rest of them: keep this PR as concise, but descriptive, as possible.._
-->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist
<!--
_Put an `x` in the boxes that apply and remove this text and the rest of them. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._
-->

- [ ] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments
<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...
-->

ciband committed:
chore: add clang format support (#43)

ciband committed:
chore: Clang Tidy Fixes (#42)

sleepdefic1t committed:
chore: add arduino sketches (#47)

This commit adds example Arduino Sketches for the ESP8266 & ESP32 boards.

ciband committed:
chore: apply clang-format (#44)

sleepdefic1t committed:
chore: update library info (#46)

sleepdefic1t committed:
chore: update library info (#48)

- add missing 'maintainer' tag.
- update PlatformIO description.

sleepdefic1t committed:
chore: update Arduino Script (#49)

- current has issues rebuilding the directories when converting back to the original structure.
This fix remedies that.

ciband committed:
chore: correct PIO builds (#52)

* chore:  Update bip39 submodule to tips

* fix:  Remove explicit install of dependencies

Allow individual projects to install dependencies.

* fix:  Add explicit version requirement for BIP39

* fix:  Disable dependency checking in PIO

PIO was getting the automated dependency checking wrong.  Turn it off in favor of being explicit

sleepdefic1t committed:
fix: rename helpers (#51)

Cpp-Crypto uses the header 'helpers/helpers.h'.
Cpp-Client also uses this 'helpers/helpers.h' naming.
These headers are not identical and contain only “helper” functions specific to its library.

When both Cpp-Client & Cpp-Crypto are included in the same project, it results in a name collision causing one of the headers to be dropped.

This PR renames 'helpers/helpers.h' to 'helpers/crypto_helpers.h’, and resolves the name collision.

sleepdefic1t committed:
fix: Remove Adapter Header (#50)

The Arduino adapter header does not solve Arduino IDE integration and is otherwise unnecessary.

This PR removes the Arduino Adapter header.

sleepdefic1t committed:
refactor: timestamps (#53)

* refactor: Slots

- add ms to Network Epochs
- refactor and fix Slot to parse Epoch ms
- update Slots tests
- update PIO 'temp_main.cpp' to set dummy board time

* fix: add IoT 'Slots::now'

* fix: add board specific code

sleepdefic1t committed:
feat: transaction to JSON (#54)

sleepdefic1t committed:
chore: Update Library Config (#55)

* misc: update library config

- list ArduinoJson as a dependency.
- add Arduino sketch path to config.
- fix the 'platformio_example' path.

* Revert "misc: update library config"

This reverts commit 55d830b821eea7854c329062c93d59f822274b4d.

* misc: update library config

- list ArduinoJson as a dependency.
- add Arduino sketch path to config.
- fix the 'platformio_example' path.

sleepdefic1t committed:
chore: Bump Version (#56)

sleepdefic1t committed:
fix: library config examples' paths (#57)

sleepdefic1t committed:
chore: remove redundant manifest fields (#58)

faustbrian committed:
chore: setup code owners (#62)

sleepdefic1t committed:
chore: Automate Arduino Script (#60)

sleepdefic1t committed:
chore: Add Arduino CircleCI Config (#61)

sleepdefic1t committed:
fix: Arduino CI (#63)

sleepdefic1t committed:
misc: Cleanup and Prep for Arduino IDE (#64)

sleepdefic1t committed:
chore: bump version (#66)

ciband committed:
fix: correct submodule ignore paths (#70)

sleepdefic1t committed:
fix: Update Build Flag (#69)

sleepdefic1t committed:
chore(release): 0.3.1 (#71)

sleepdefic1t committed:
test: use core fixtures (#74)

ghost commented:
The [ci/circleci: build-linux-clang-5](https://circleci.com/gh/ArkEcosystem/cpp-crypto/767?utm_campaign=vcs-integration-link&amp;utm_medium=referral&amp;utm_source=github-build-link) job is failing as of [a1b0629341c94ac78c1d0f8a4ffcafb58e55e9a3](https://github.com/ArkEcosystem/cpp-crypto/commit/a1b0629341c94ac78c1d0f8a4ffcafb58e55e9a3). Please [review the logs for more information](https://circleci.com/gh/ArkEcosystem/cpp-crypto/767?utm_campaign=vcs-integration-link&amp;utm_medium=referral&amp;utm_source=github-build-link).

_Once you've pushed the fixes, the build will automatically re-run. Thanks!_

<!-- ACTION:REPORT_FAILING_TESTS -->

sleepdefic1t committed:
chore: Update ArduinoJson Package (#76)

sleepdefic1t committed:
fix: improve Windows support. (#83)

Windows doesn't handle dynamic allocation well.

- transaction.cpp
  - explicitly define est. doc max size of 900.
  - No need for string since we're stating size.
- message.cpp
  - let string handle doc size allocation.

sleepdefic1t committed:
chore: disable PlatformIO ESP8266 Tests (#87)

AUnit has stopped correctly building tests for the partially-supported ESP8266 platform.
It seems this may be a change in ESP8266 Core, so this PR disables ESP8266 tests in PlatformIO until support is worked out.

sleepdefic1t committed:
feat: Increase VendorField Size (#84)

sleepdefic1t committed:
fix: 0 ARKtoshi amount handling (#85)

sleepdefic1t committed:
chore(ci): setup Appveyor (#89)

sleepdefic1t committed:
refactor: Use BIP66 Library (#88)

sleepdefic1t committed:
refactor: Remove bip39 library (#86)

sleepdefic1t committed:
refactor(slots): Improve Slots Implementations (#92)

- Further decouple OS code from IoT code.
- Add missing ESP8266 functionality.
- Allow removal of lib/STL dependency.
  - STL lib is only included to provide a conv. method for IoT to use date.h
  - date.h is now only used by OS builds, which will have the STL by default.
- update changelog.
- add PIO filters for lib/date.

faustbrian committed:
chore: use organization-wide GitHub Configuration (#94)

sleepdefic1t committed:
style: Improve Formatting and Maintainability (#98)

per1234 committed:
chore: use correct field separator in keywords.txt (#96)

The Arduino IDE requires the use of a single true tab separator between the keyword name and identifier. When spaces are used rather than a true tab, the keyword is not highlighted.

Reference:
https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5:-Library-specification#keywords

sleepdefic1t committed:
docs: update documentation link (#93)

sleepdefic1t committed:
build(platformio): Improve PlatformIO Configuration (#101)

sleepdefic1t committed:
test(networks): Fix AbstractNetwork Tests (#102)

Current tests are ran in different orders per platform.
This led to a modified global test custom_network being modified improperly.

This PR properly handles that order variation by creating a local object to be modified.

In future iterations, an AbstractNetwork will allow `const`.

sleepdefic1t committed:
fix(transaction): Improve Numeric Json Serialization (#103)

Current implementation sometimes fails to properly serialize numbers in the `toJson()` method.
This is especially detrimental to some Linux builds and all Arduino IDE builds (Arduino IDE Library version (Cpp-Crypto-Arduino-v.0.3.0)).

Specifically, this PR does the following:
- moves `<cinttypes>` to `crypto_helpers.h` for non-IoT builds.
- forces type serialization for:
  - `amount` (llu/uint64_t).
  - `fee` (llu/uint64_t).
  - `network` (uint8_t/int).
  - `timestamp` (lu/uint32_t).
  - `type` (uint8_t/int).
  - `version` (uint8_t/int).
- updates the changelog.

sleepdefic1t committed:
refactor(configuration): Support Bridgechains & Custom Configurations (#105)

The current implementation only allows use with Devnet unless values are manually changed in `/src`.

This PR enables network configuration via the public api.
(Devnet, Mainnet, Testnet, and Custom Networks (bridgechains).

Specifically this PR does the following:
- Creates a Configuration class.
- Creates Fee and Network Manager children for the Configuration class.
- Allows passing a `Configuration to` Transaction Builder with a default value of `Devnet` & `StaticFees` (non-breaking).
- Adds a FeePolicy-type.
- Adds a Fee class (container for fee policies).
- Adds a Networks class (container for preset networks).
- Updates the TransactionTypes enum.
- Improves the `Network` class and Network implementions.
- Updates the `Slot` class to reflect changes (non-breaking).
- Updates Transaction-related classes to reflect updates (non-breaking).
- Adds tests for all changes.
- Updates the Arduino IDE script.
- Updates `.ino` Arduino sketches with Configuration examples.
- Updates documentation with Configuration examples.
- Updates the `keywords.txt` file to reflect changes.
- Updates the changelog.
- Updates version to `v.0.6.0`.

sleepdefic1t committed:
chore(manifest): update dependencies (#108)

sleepdefic1t committed:
fix(manifest): Correct Supported Platforms (#107)

The current library manifest uses a wildcard `*` for the `platforms` value incorrectly signifying that it supports all platforms.

This is incorrect as the Cpp SDK's only officially support the espressif8266 & espressif32 platforms.  This PR corrects this discrepancy.

sleepdefic1t committed:
chore(transaction): update amounts json (#111)

sleepdefic1t committed:
Merge branch 'master' into chore/update-changelog-and-merge-#93-from-master

sleepdefic1t committed:
chore: update changelog and merge #93 from master

sleepdefic1t committed:
Update CHANGELOG.md

faustbrian committed:
chore: update changelog and merge #93 from master (#113)

faustbrian committed:
chore: merge develop into master (#115)

faustbrian committed:
chore: setup probot/stale configuration (#117)

faustbrian committed:
chore(readme): add lead maintainer information (#118)
